lib/vrl/stdlib/src/decode_mime_q.rs

```diff
@@ -37,67 +37,70 @@ impl Function for DecodeQ {
     ) -> Compiled {
         let value = arguments.required("value");
 
-        Ok(DecodeQFn { value }.as_expr())
+        Ok(DecodeMimeQFn { value }.as_expr())
     }
 
     fn examples(&self) -> &'static [Example] {
         &[
             Example {
                 title: "Single",
-                source: r#"decode_q!("=?utf-8?b?SGVsbG8sIFdvcmxkIQ==?=")"#,
+                source: r#"decode_mime_q!("=?utf-8?b?SGVsbG8sIFdvcmxkIQ==?=")"#,
                 result: Ok(r#"Hello, World!"#),
             },
             Example {
                 title: "Embedded",
-                source: r#"decode_q!("From: =?utf-8?b?SGVsbG8sIFdvcmxkIQ==?= <=?utf-8?q?hello=5Fworld=40example=2ecom?=>")"#,
+                source: r#"decode_mime_q!("From: =?utf-8?b?SGVsbG8sIFdvcmxkIQ==?= <=?utf-8?q?hello=5Fworld=40example=2ecom?=>")"#,
                 result: Ok(r#"From: Hello, World! <hello_world@example.com>"#),
             },
             Example {
                 title: "Without charset",
-                source: r#"decode_q!("?b?SGVsbG8sIFdvcmxkIQ==")"#,
+                source: r#"decode_mime_q!("?b?SGVsbG8sIFdvcmxkIQ==")"#,
                 result: Ok(r#"Hello, World!"#),
             },
         ]
     }
 }
 
 #[derive(Clone, Debug)]
-struct DecodeQFn {
+struct DecodeMimeQFn {
     value: Box<dyn Expression>,
 }
 
-impl FunctionExpression for DecodeQFn {
+impl FunctionExpression for DecodeMimeQFn {
     fn resolve(&self, ctx: &mut Context) -> Resolved {
         let value = self.value.resolve(ctx)?;
 
-        decode_q(value)
+        decode_mime_q(value)
     }
 
     fn type_def(&self, _: &state::TypeState) -> TypeDef {
         TypeDef::bytes().fallible()
     }
 }
 
-fn decode_q(bytes: Value) -> Resolved {
+fn decode_mime_q(bytes: Value) -> Resolved {
     // Parse
     let input = bytes.try_bytes_utf8_lossy()?;
     let input: &str = &input;
     let (remaining, decoded) = alt((
         fold_many1(
             parse_delimited_q,
-            || Ok(String::new()),
+            || Result::<String>::Ok(String::new()),
             |result, (head, word)| {
                 let mut result = result?;
 
                 result.push_str(head);
-                if let Some(word) = word {
-                    result.push_str(&word.decode_word()?);
-                }
+                result.push_str(&word.decode_word()?);
 
                 Ok(result)
             },
         ),
-        map(parse_internal_q, |word| word.decode_word()),
+        alt((
+            map_opt(parse_internal_q, |word| {
+                word.decode_word().map(|word| Ok(word)).ok()
+            }),
+            success(Ok(String::new())),
```

Comment:
Looks like CI has flagged this as having a redundant closure. `make check-clippy` should show you it locally.